test/test_response.py

```diff
@@ -377,6 +377,59 @@ def test_empty_stream(self):
 
         self.assertRaises(StopIteration, next, stream)
 
+    def test_length_no_header(self):
+        fp = BytesIO(b'12345')
+        resp = HTTPResponse(fp, preload_content=False)
+        assert resp.length is None
+
+    def test_length_w_valid_header(self):
+        headers = {"content-length": "5"}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=headers, preload_content=False)
+        assert resp.length == 5
+
+    def test_length_w_bad_header(self):
+        garbage = {'content-length': 'foo'}
+        fp = BytesIO(b'12345')
+
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+        garbage['content-length'] = "-10"
+        resp = HTTPResponse(fp, headers=garbage, preload_content=False)
+        assert resp.length is None
+
+    def test_length_when_chunked(self):
+        headers = {'content-length': '5',
+                   'transfer-encoding': 'chunked'}
```

Comment:
I added this test because the initial `httplib` length logic checks to make sure things aren't chunked. I view this test in the same vein as receiving a Content-Length of "foo". It shouldn't happen but do we want to actually return the content length in the event urllib3 happens to receive both? I'd say no, the length should be None because it can't be determined.

Alternatively, we could raise an exception here but I'm not sure if that would buy us anything useful other than aborting the operation.